tracer/src/Datadog.Trace/Iast/Aspects/System.Text/StringBuilderAspects.cs

```diff
@@ -0,0 +1,199 @@
+        return result;
+    }
+
+    /// <summary>  StringBuilder.Append aspect </summary>
+    /// <param name="target"> StringBuilder instance </param>
+    /// <param name="value"> value parameter </param>
+    /// <returns> instance.Append() </returns>
+    [AspectMethodReplace("System.Text.StringBuilder::Append(System.String)", AspectFilter.StringLiteral_1)]
+    public static StringBuilder Append(StringBuilder target, string? value)
+    {
+        var initialLength = target.Length;
+        var length = value?.Length ?? 0;
+        return StringBuilderModuleImpl.OnStringBuilderAppend(target.Append(value), initialLength, value, length, 0, length);
+    }
+
+#if !NETFRAMEWORK
+    /// <summary>  StringBuilder.Append aspect </summary>
+    /// <param name="target"> StringBuilder instance </param>
+    /// <param name="value"> value parameter </param>
+    /// <returns> instance.Append() </returns>
+    [AspectMethodReplace("System.Text.StringBuilder::Append(System.Text.StringBuilder)")]
+    public static StringBuilder Append(StringBuilder target, StringBuilder? value)
+    {
+        var initialLength = target.Length;
+        var length = value?.Length ?? 0;
+        return StringBuilderModuleImpl.OnStringBuilderAppend(target.Append(value), initialLength, value, length, 0, length);
+    }
+#endif
+
+    /// <summary>  StringBuilder.Append aspect </summary>
+    /// <param name="target"> StringBuilder instance </param>
+    /// <param name="value"> string parameter </param>
+    /// <param name="startIndex"> startIndex parameter </param>
+    /// <param name="count"> count parameter </param>
+    /// <returns> instance.Append() </returns>
+    [AspectMethodReplace("System.Text.StringBuilder::Append(System.String,System.Int32,System.Int32)", AspectFilter.StringLiteral_1)]
+    public static StringBuilder Append(StringBuilder target, string? value, int startIndex, int count)
+    {
+        var initialLength = target.Length;
+        return StringBuilderModuleImpl.OnStringBuilderAppend(target.Append(value, startIndex, count), initialLength, value, value?.Length ?? 0, startIndex, count);
+    }
+
+#if !NETFRAMEWORK
+    /// <summary>  StringBuilder.Append aspect </summary>
+    /// <param name="target"> StringBuilder instance </param>
+    /// <param name="value"> string parameter </param>
+    /// <param name="startIndex"> startIndex parameter </param>
+    /// <param name="count"> count parameter </param>
+    /// <returns> instance.Append() </returns>
+    [AspectMethodReplace("System.Text.StringBuilder::Append(System.Text.StringBuilder,System.Int32,System.Int32)", AspectFilter.StringLiteral_1)]
+    public static StringBuilder Append(StringBuilder target, StringBuilder? value, int startIndex, int count)
+    {
+        var initialLength = target.Length;
+        return StringBuilderModuleImpl.OnStringBuilderAppend(target.Append(value?.ToString(), startIndex, count), initialLength, value, value?.Length ?? 0, startIndex, count);
```

Comment:
Aren't you changing the behaviour here by passing a string instead of a stringbuilder? I mean all in all it's the same but I thought we didn't want to do that kind of things and call the exact same method

Reply:
This particular aspect covers is a very particular and not ideal situation. The problem is that in netcoreapp2.1 and above, we have this method overload defined, but our tracer library runs netstandard when called from netcoreapp2.1 and in netstandard, it is not defined, so we cannot call the original method. I have added a comment, which might be helpful. Also, we can limit this effect with a precompiled directive.

The choice here is to exclude the instrumentation of this method overload in the netstandard library or call ToString()

Reply:
Ok I just wanted to make sure it was on purpose. What's the biggest risk? I haven't look at that implementation? I assume the biggest risk is to allocate one extra string each time this method is called, right? 
Also, where's the comment?

Reply:
I forgot to push my commit and left it pending!!!

The implementation of this method relies on StringBuilder.CopyTo(). The result should be the same, with some minor differences in terms of performance and, as you mention, allocation. I think that there should be no problem. We used this approach in Hdiv for this overload and had no problems with it.

Still, not supporting coverage for this overload in netcore2.1 would have minimum impact on our capacity to taint.